technical-reports/format/composite-types.md

```diff
@@ -60,155 +75,205 @@ Other examples might be:
+  - Tools MAY provide specialised functionality for composite tokens. For example, a design tool may let the user pick from a list of all available shadow tokens when applying a drop shadow effect to an element.
+
+## Border
 
-## Type checking
+Represents a border style. The type property must be set to the string “border”. The value must be an object with the following properties:
 
-Just as with "normal" types for tokens, using a custom, composite type will allow tools to check that the values you use match the expected type. In our color pair example above, attempting to do the following would be invalid and tools should ignore the token and show an error to the user, since "Comic Sans MS" is not a valid color value.
+- `color`: The color of the border. The value of this property must be a valid [color value](#color) or a reference to a color token.
+- `width`: The width or thickness of the border. The value of this property must be a valid [dimension value](#dimension) or a reference to a dimension token.
+- `style`: The border's style, for example "solid" or "dashed". The value of this property must be a valid JSON string or a reference to a string token.
 
-<aside class="example" title="Invalid type">
+<aside class="example" title="Border composite token examples">
 
 ```json
 {
-  "broken-token": {
-    "type": "{my types.color pair}",
-    "value": {
-      "foreground": "Comic Sans MS",
-      "background": "#eeeeee"
+  "border": {
+    "heavy": {
+      "type": "border",
+      "value": {
+        "color": "#36363600",
+        "width": "3px",
+        "style": "solid"
+      }
+    },
+    "focusring": {
+      "type": "border",
+      "value": {
+        "color": "{color.focusring}",
+        "width": "1px",
+        "style": "dashed"
+      }
     }
   }
 }
 ```
 
 </aside>
 
-Likewise, IDEs can offer appropriate auto-completions to users that manually author design token files.
+## Transition
+
+TO-DO
 
-## Tool support
+## Shadow
 
-Since composite types are user-defined, tools cannot provide specialized handling of them since they have no advanced knowledge of the type declarations. Here are some suggested strategies to get the most value out of user-defined composite types:
+Represents a shadow style. The type property must be set to the string “shadow”. The value must be an object with the following properties:
 
-### Fallbacks
+- `color`: The color of the shadow. The value of this property must be a valid [color value](#color) or a reference to a color token.
+- `x`: The horizontal offset that shadow has from the element it is applied to. The value of this property must be a valid [dimension value](#dimension) or a reference to a dimension token.
+- `y`: The vertical offset that shadow has from the element it is applied to. The value of this property must be a valid [dimension value](#dimension) or a reference to a dimension token.
```

Comment:
Although this might be more verbose, should we call `x` and `y` `offsetX` and `offsetY` respectively? The CSS spec and iOS use the term `offset`:
* https://drafts.csswg.org/css-backgrounds/#vertical-offset
* https://developer.mozilla.org/en-US/docs/Web/CSS/box-shadow#values
* https://developer.apple.com/documentation/quartzcore/calayer/1410970-shadowoffset

Reply:
Done